crates/oxc_span/src/source_type/mod.rs

```diff
@@ -269,68 +265,50 @@ impl SourceType {
     }
 
     #[must_use]
-    pub const fn with_script(mut self, yes: bool) -> Self {
```

Comment:
Question: should we support both setter and builder methods? Source type is a user-facing API, and users may want one or the other depending on their use case